lib/hashes.md

```diff
@@ -5,40 +5,97 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+I know you aren't looking for code.
+
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+puts StateCaps2.keys.inspect
+puts StateCaps2.values.inspect
+puts StateCaps2["Vermont"] #This returns just the capital of Vermont, which is "Montpelier.
+
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+There are probably several ways to do this, but I found a Merge method that works kind of like "join".
+I made a second hash of more states and capitals, then used the merge! method to make one larger hash.
+This tacked on the new variables at visually at least is the "end" of the old hash.
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+StateCaps3 = {"New York"=>"Albany", "Texas"=>"Austin", "Kentucky"=>"Frankfort"}
+StateCaps2.merge! StateCaps3
+puts StateCaps2
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This seems like something that calls for a loop, specifically an "each do end" loop.
+The bit of code below is an adaptation of found material, which isn't my favorite way to do this, 
+and I'm still working backwards a little bit to say what everything does. Mostly I am taking out
+pieces that I don't understand and seeing what error message I get. I get the impression that 
+the "#" before {key} and {value} tells Ruby to evaluate each key and value in the hash and loop them into the appropriate places
+one at a time. The block | | thing is starting to make sense. Need to do a lot more coding before any of it is like English.
 
-# How would you change the value of a particular element in a Hash?
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines", "New York"=>"Albany", "Texas"=>"Austin", "Kentucky"=>"Frankfort"}
+StateCaps2.each do 
+    |key, value| puts "The capital of #{key} is #{value}."
+end
 
-Replace me with your response...
+
+# How would you change the value of a particular element in a Hash?
+There is probably a much more elegant way to do this than what I have done here. Using the states and capitals hash again, I changed
+the capital of Oregon from Salem to Portland. It's not right, but let's say it is. It looks like the merge found the 
+key "Oregon" and over-wrote the old capital "Salem" with the new one, "Portland." There is probably a way to do this
+in one line of code.
+
+PROGRAM TO CHANGE THE VALUE OF AN ELEMENT IN A HASH. let's say Oregon moved its capital to Portland and we had to update records.
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines", "New York"=>"Albany", "Texas"=>"Austin", "Kentucky"=>"Frankfort"}
+StateCaps3 = {"Oregon"=>"Portland"}
+StateCaps2.merge!(StateCaps3) 
```

Comment:
Same note as above, regarding a more precise way to change the value of a single element in a Hash.